engine/src/main/java/org/terasology/rendering/nui/widgets/UISlider.java

```diff
@@ -173,14 +171,39 @@ public String getMode() {
             return DISABLED_MODE;
         }
 
-        if (active) {
+        if (active || (TabbingManager.focusedWidget != null && TabbingManager.focusedWidget.equals(this))) {
             return ACTIVE_MODE;
         } else if (tickerListener.isMouseOver()) {
             return HOVER_MODE;
         }
         return DEFAULT_MODE;
     }
 
+    private void changeValue(float delta) {
+        float newValue = TeraMath.clamp(getValue() + delta, 0, getRange() + getMinimum());
+        setValue(newValue);
+    }
+
+    @Override
+    public boolean onKeyEvent(NUIKeyEvent event) {
+        if (event.isDown() && TabbingManager.focusedWidget.equals(this)) {
```

Comment:
See my above point about null checks. If you've got one above it's probably good to have one here.